app/models/runs_extractors.rb

```diff
@@ -60,15 +64,31 @@ def extract(classification, and_reduce: false)
 
     if and_reduce
       extracts = extracts.select { |extract| extract != Extractor::NoData }
+      return unless extracts.present?
 
-      if extracts.present?
-        ids = extracts.map(&:id)
-        ReduceWorker.perform_async(classification.workflow_id, "Workflow", classification.subject_id, classification.user_id, ids)
+      ids = extracts.map(&:id)
 
-        project = Project.find_by_id(workflow.project_id)
-        if project && project.has_reducers?
-          ReduceWorker.perform_async(classification.project_id, "Project", classification.subject_id, classification.user_id, ids)
+      if workflow && workflow.reducers.any?
+        worker = if workflow.has_external_reducers?
+          ReduceWorkerExternal
+        else
+          ReduceWorker
         end
+
+        worker.set(queue: workflow.custom_queue_name.to_sym) if workflow.custom_queue_name.present?
+        worker.perform_async(classification.workflow_id, 'Workflow', classification.subject_id, classification.user_id, ids)
+      end
+
+      project = Project.find_by_id(classification.project_id)
+      if project && project.has_reducers?
```

Comment:
use safe navigation operator
```suggestion
      if project&.has_reducers?
```

Reply:
yep